lib/utils.js

```diff
@@ -203,6 +203,56 @@ function rgbToXY(red, green, blue) {
     return {x: Number.parseFloat(x), y: Number.parseFloat(y)};
 }
 
+function xyToRGB(x, y, brightness) {
+    // Set to maximum brightness if no custom value was given (Not the slick ECMAScript 6 way for compatibility reasons)
+    if (brightness === undefined) {
+        brightness = 254;
+    }
+
+    const z = 1.0 - x - y;
+    const Y = (brightness / 254).toFixed(2);
+    const X = (Y / y) * x;
+    const Z = (Y / y) * z;
+
+    // Convert to RGB using Wide RGB D65 conversion
+    let red = X * 1.656492 - Y * 0.354851 - Z * 0.255038;
+    let green = -X * 0.707196 + Y * 1.655397 + Z * 0.036152;
+    let blue = X * 0.051713 - Y * 0.121364 + Z * 1.011530;
+
+    // If red, green or blue is larger than 1.0 set it back to the maximum of 1.0
+    if (red > blue && red > green && red > 1.0) {
+        green = green / red;
+        blue = blue / red;
+        red = 1.0;
+    } else if (green > blue && green > red && green > 1.0) {
+        red = red / green;
+        blue = blue / green;
+        green = 1.0;
+    } else if (blue > red && blue > green && blue > 1.0) {
+        red = red / blue;
+        green = green / blue;
+        blue = 1.0;
+    }
+
+    // Reverse gamma correction
```

Comment:
Should this not also use the fancy hue variant done by gammaCorrectRGB ?

Reply:
Which variant do you mean?

Reply:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/d751f07919cc1122da3f0ac75053a7ba0cfe137b/lib/utils.js#L184 was based on the Signify Hue dev docs right... should we move this to a seperate reverse function too reverse this correction?

Reply:
Ah didn't know it was hue, but seems so: https://github.com/usolved/cie-rgb-converter/blob/79a9eb81abfd8126a94ff7bee69fc8daaf79d6e1/cie_rgb_converter.js#L4

But this should be the reverse of it (both functions are taken from `cie_rgb_converter.js`)